### PR TITLE
feat(browser): #MCLOUD-5407 fileUploadWithToken suggestion

### PR DESCRIPTION
**Pull request by avrahamr**
Implementing the form on our own has become too complicated, so may I suggest something like this?
If you agree with the concept I can continue with the necessary tests and will listen to any code review suggestions

### CODE REVIEW

**Comment by lelikg** [comment]
This looks good, but it seems that you duplicated the logic from inside the uploadFile function. We'd rather avoid code duplications, perhaps it's possible to refactor the uploadFile function and extract the uploadFileWithToken part from there?

Tests would be welcome obviously, but if you wish I can make this change on my own next week.

**Comment by avrahamr** [comment]
agreed about the refactoring on code duplication, in my defense for the purpose of just asking about the concept it was too much complicated :-)

I will be great if you can take it next week.
 
LMK if you need me to open an issue/jira.

Thanks a lot

File: src/platform/management/file-uploader.ts [hunk 1]
```diff
@@ -165,6 +173,45 @@ export class FileUploader implements IFileUploader {
...
+   * @param {UploadFileRequest?} uploadFileRequest
+   */
+  uploadFileWithToken(
+    path: string,
+    file: string | Buffer | Stream,
+    uploadToken: string,
+    uploadUrl: string,
+    uploadFileRequest?: UploadFileRequest,
+  ) {
+    let stream, size, streamErrorPromise;
```
Comment (davydof) [inline src/platform/management/file-uploader.ts]: you don't use `size`

File: src/platform/management/file-uploader.ts [hunk 2]
```diff
@@ -165,6 +173,45 @@ export class FileUploader implements IFileUploader {
...
+      ...uploadFileRequest,
+    };
+
+    return Promise.race([
+      this.uploadFileWithPost(uploadUrl, form),
+      streamErrorPromise,
+    ]).then(({ payload }) => {
+      return [new FileDescriptor(payload)];
+    });
+  }
```
Comment (davydof) [inline src/platform/management/file-uploader.ts]: what about tests?

**Comment by avrahamr** [comment]
@davydof I guess we need to set expectations. I only used this PR as a means of expressing what is needed from the API, I didn't figure the proper implementation and the needed tests. Fell free to close it and start fro scratch. In any case I opened it:
https://jira.wixpress.com/browse/MCLOUD-5407  
Hope you can get to it soon :-)

**Comment by davydof** [comment]
@avrahamr how is it important for you? is it blocker?
I can plan it to my next sprint(next week)

**Comment by avrahamr** [comment]
@davydof it's important, but not critical or urgent. Next sprint or two is great.

**Pull request merged**